packages/eslint-plugin/configs/react.js

```diff
@@ -12,6 +12,9 @@ module.exports = {
 		'react-hooks',
 	],
 	rules: {
+		'@wordpress/no-unused-vars-before-return': [ 'error', {
+			excludePattern: '^use',
```

Comment:
Does this pattern work for all kind of things or only function calls? Maybe the naming should indicate that it's about function calls?

Reply:
> Does this pattern work for all kind of things or only function calls? Maybe the naming should indicate that it's about function calls?

Today, the rule is only concerned with function calls, as a rough trigger for what should be considered expensive:

https://github.com/WordPress/gutenberg/blob/e65c55d4c90ae862236bda5fcf192b486496acc1/packages/eslint-plugin/rules/no-unused-vars-before-return.js#L24-L25

I suppose this doesn't always need to be the case, and in the future we could be more inclusive on what we consider to be "expensive". We can be more open in this module to future breaking changes, but I think it's a sensible change to make even now, in line with what the rule is intended to enforce.

Reply:
Yes, not a big concern.

Reply:
Hm, let's leave it as is for now. For the moment, the description of the rule is pretty explicit about this applying to functions:

>To avoid wastefully computing the result of a function call when assigning a variable value

...and I'm not inclined to think this will change anytime soon.